hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OMDBCheckpointServlet.java

```diff
@@ -644,29 +646,42 @@ public BootstrapStateHandler.Lock getBootstrapStateLock() {
   }
 
   static class Lock extends BootstrapStateHandler.Lock {
-    private final BootstrapStateHandler keyDeletingService;
-    private final BootstrapStateHandler sstFilteringService;
-    private final BootstrapStateHandler rocksDbCheckpointDiffer;
-    private final BootstrapStateHandler snapshotDeletingService;
+    private final Optional<BootstrapStateHandler> keyDeletingService;
+    private final Optional<BootstrapStateHandler> sstFilteringService;
+    private final Optional<BootstrapStateHandler> rocksDbCheckpointDiffer;
+    private final Optional<BootstrapStateHandler> snapshotDeletingService;
     private final OzoneManager om;
 
     Lock(OzoneManager om) {
+      Preconditions.checkNotNull(om);
+      Preconditions.checkNotNull(om.getKeyManager());
+      Preconditions.checkNotNull(om.getMetadataManager());
+      Preconditions.checkNotNull(om.getMetadataManager().getStore());
+
       this.om = om;
-      keyDeletingService = om.getKeyManager().getDeletingService();
-      sstFilteringService = om.getKeyManager().getSnapshotSstFilteringService();
-      rocksDbCheckpointDiffer = om.getMetadataManager().getStore()
-          .getRocksDBCheckpointDiffer();
-      snapshotDeletingService = om.getKeyManager().getSnapshotDeletingService();
+      keyDeletingService = Optional.ofNullable(om.getKeyManager().getDeletingService());
+      sstFilteringService = Optional.ofNullable(om.getKeyManager().getSnapshotSstFilteringService());
+      rocksDbCheckpointDiffer = Optional.ofNullable(om.getMetadataManager().getStore()
+          .getRocksDBCheckpointDiffer());
+      snapshotDeletingService = Optional.ofNullable(om.getKeyManager().getSnapshotDeletingService());
     }
 
     @Override
     public BootstrapStateHandler.Lock lock()
         throws InterruptedException {
       // First lock all the handlers.
-      keyDeletingService.getBootstrapStateLock().lock();
-      sstFilteringService.getBootstrapStateLock().lock();
-      rocksDbCheckpointDiffer.getBootstrapStateLock().lock();
-      snapshotDeletingService.getBootstrapStateLock().lock();
+      if (keyDeletingService.isPresent()) {
```

Comment:
Can changed to `ifPresent`.

Reply:
Since `lock` throws `InterruptedException` (unlike `unlock`), it might not be possible to use `ifPresent`.